src/aspire/utils/rotation.py

```diff
@@ -0,0 +1,280 @@
+        self.shape = (num_rots, 3, 3)
+
+    def __str__(self):
+        """
+        String representation.
+        """
+        return (
+            f"Rotation object with matrices[{self.num_rots}, 3, 3] of {self.dtype} type"
+        )
+
+    def __len__(self):
+        return self.num_rots
+
+    def __getitem__(self, item):
+        return self.data[item]
+
+    def __setitem__(self, key, value):
+        self.data[key] = value
+
+    def __mul__(self, other):
+        if isinstance(other, Rotation):
+            output = self.data @ other.data
+        else:
+            output = self.data * other
+        return output
+
+    def _uniform_random_angles(self, n, seed=None):
+        """
+        Generate random 3D rotation angles in degrees
+
+        :param n: The number of rotation angles to generate
+        :param seed: Random integer seed to use. If None,
+            the current random state is used.
+        :return: A n-by-3 ndarray of rotation angles in degrees
+        """
+        # Generate random rotation angles, in radians
+        angles = np.zeros((n, 3), dtype=self.dtype)
+        with Random(seed):
+            angles = np.column_stack(
+                (
+                    np.random.random(n) * 2 * np.pi,
+                    np.arccos(2 * np.random.random(n) - 1),
+                    np.random.random(n) * 2 * np.pi,
+                )
+            )
+        # Return random rotation angles in degrees
+        return np.rad2deg(angles).astype(self.dtype)
+
+    @property
+    def rot_matrices(self):
+        """
+        :return: Rotation matrices as a n x 3 x 3 array
+        """
+        return self._rotations.as_matrix().astype(self.dtype)
+
+    @rot_matrices.setter
+    def rot_matrices(self, values):
+        """
+        Set rotation matrices
+
+        :param values: Rotation matrices as a n x 3 x 3 array
+        :return: None
+        """
+        self._rotations = sp_rot.from_matrix(values.astype(self.dtype))
+        self.data = values.astype(self.dtype)
+
+    @property
+    def angles(self):
+        """
+        :return: Rotation angles in degrees, as a n x 3 array
+        """
+        return self._rotations.as_euler(self.rot_seq, degrees=True).astype(self.dtype)
+
+    @angles.setter
+    def angles(self, values):
+        """
+        Set rotation angles in degrees
+
+        :param values: Rotation angles in degrees, as a n x 3 array
+        :return: None
+        """
+        self._rotations = sp_rot.from_euler(
+            self.rot_seq, values.astype(self.dtype), degrees=True
+        )
+
+    @property
+    def T(self):
+        """
+        Shortcut to get full set of transposed rotation matrices
+        """
+        return self.transpose()
+
+    def transpose(self):
+        """
+        Apply transpose operation to all rotation matrices
+
+        :return: The set of transposed matrices
+        """
+        return np.transpose(self.data, axes=(0, 2, 1))
+
+    def asnumpy(self):
```

Comment:
Isn't this the same as `rot_matrices` above?

Reply:
Tried to make a consistent interface with Image and Volume class which can return numpy array. `rot_matrices` was replaced by  `matrices` in new commit.